server/src/dependency/heed.rs

```diff
@@ -63,11 +68,11 @@ impl<'db> umi_app::Dependencies<'db> for HeedDependencies {
     type TransactionRepository = transaction::HeedTransactionRepository<'db>;
 
     fn block_hash_lookup(&self) -> Self::BlockHashLookup {
-        SharedBlockHashCache::new()
+        self.block_hash_lookup.clone()
```

Comment:
So the crux of this bug was that a single block hash cache instance should have been shared between the lookup and reader, but they were two separate unsynchronized ones, I suppose.